### ISSUE TITLE
Functional Requirements: Save Login Credentials and Verification when Opening the App

### ISSUE THREAD

**Issue by iantrieschmann** [verdigado/gruene_app#204]
**Der Login bleibt auf dem Gerät gespeichert, User:innen sind dauerhaft eingeloggt. Die Notwendigkeit eines erneuten Einloggens sollte wann immer möglich verhindert werden.**

Daher müssen Nutzer:innen beim Öffnen der Applikation aus Sicherheitsgründen verifiziert werden. Die Verifizierung sollte über eine möglichst einfache, barrierearme und schnelle Weise erfolgen, gerne z.B. via Face-ID, Fingerabdruck, kurzer PIN, evtl. könnte hier die Gerätevoreinstellung genutzt werden. Wenn nötig muss hierfür noch eine Einstellung / Abfrage innerhalb der App entwickelt werden.

**Comment by JuliaFreitagBGSt** [comment]
Question: Automatic login should expire after a certain period of time. Suggestion: 90 days. Is that common? What is recommended?

**Comment by JuliaFreitagBGSt** [comment]
@maxpfe as mentioned before, issue is not ready for development until questions above are clarified
pls clarify or otherwise issue stays in backlog

**Comment by steffenkleinle** [comment]
@JuliaFreitagBGSt can you please clarify what needs to be done here? I think login information are already saved?

**Comment by JuliaFreitagBGSt** [comment]
> [@JuliaFreitagBGSt](https://github.com/JuliaFreitagBGSt) can you please clarify what needs to be done here? I think login information are already saved?

Yes, but we still need biometric methods (face ID, fingerprint, short PIN or longer password), so user easily access the App

**Comment by steffenkleinle** [comment]
> > [@JuliaFreitagBGSt](https://github.com/JuliaFreitagBGSt) can you please clarify what needs to be done here? I think login information are already saved?
> 
> Yes, but we still need biometric methods (face ID, fingerprint, short PIN or longer password), so user easily access the App

You mean users should have to authenticate themselves every time they open the app? I don't think that is necessarily `easier` then it is now without.

**Comment by volkramweber** [comment]
@maxpfe @JuliaFreitagBGSt 
So the app should ask for local auth (biometric or pin) every time you open it?
Or only if you open the app and are still logged in?

**Comment by JuliaFreitagBGSt** [comment]
> [@volkramweber](https://github.com/volkramweber) The biometric should be able to replace the login via name, password + 2fa. After I entered all my login information once (first time using the app) I should be able to replace it via biometric or my phone pin, if I want. When I am logged in, I am logged in and it stays like that - no biometric, no other login. When I log out and log in again, I should be able to use the biometric I set up at first instead of the longer login option via name etc.

Please implement @maxpfe @volkramweber

**Comment by JuliaFreitagBGSt** [comment]
Result of discussion with @maxpfe 

We will **not deal with this topic further in the app project**, but will generally consider it in further Keycloak developments.

In the meantime, we are following your recommendation @svenseeberg and hereby instruct that the **session duration of the app be set to one year** (365 days). Only if users deliberately log out or reinstall the app will it be necessary to log in again.

**Comment by svenseeberg** [comment]
> session duration of the app be set to one year (365 days).

This definition is a bit unclear, because are 2 "different session durations":
1. maximum session time: the user is definitely logged out after that period. That means 365 days after having logged in, the session is terminated and the user has to re-authenticate.
2. session idle time: a period the user can not interact with the app while still being logged in. This is what I intended and I would then recommend to set "no limit" (practically maybe 5 years) to the max session time.

**Comment by JuliaFreitagBGSt** [comment]
> 2\. a period the user can not interact with the app while still being logged in.

@svenseeberg Sorry, I don't get that. Why should user not be able to interact with app while beeing logged in?

**Comment by svenseeberg** [comment]
The idle time is the time the session is kept active while the user is not using the app. Lets have 2 examples here for an idle time of 365 days:
- When I do not use/open the app for 363 days and I open it on the 364th day, I'm still logged on. This will then again refresh the session and it will be valid for another 365 days.
- When I do not use/open for 366 days, I'm logged out.

So with the limits I suggested users can stay logged in if they open the app once a year. But after 5 years the session will be terminated because that will be our maximum session time.
